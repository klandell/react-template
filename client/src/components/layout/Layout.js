import React, { Component } from 'react';
import { func } from 'prop-types';

export default class Layout extends Component {
    static propTypes = {
        onAboutClick: func.isRequired,
        onNotFoundClick: func.isRequired,
    }
    render() {
        const { children, onAboutClick, onNotFoundClick } = this.props;

        return (
            <div className="layout">
                <button className="about-btn" onClick={onAboutClick}>About</button>
                <button className="not-found-btn" onClick={onNotFoundClick}>404 Not Found</button>
                {children}
            </div>
        );
    }
}
