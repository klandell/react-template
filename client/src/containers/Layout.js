import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { withRouter } from 'react-router';
import { showAbout, showNotFound } from 'Actions/layoutActions';

@connect(state => state, dispatch => ({
    actions: {
        showAbout,
        showNotFound,
    },
    dispatch,
}))
class Layout extends React.Component {
    static propTypes = {
        actions: PropTypes.shape({
            showAbout: PropTypes.func.isRequired,
            showNotFound: PropTypes.func.isRequired,
        }).isRequired,
        dispatch: PropTypes.func.isRequired,
    }

    onAboutClick() {
        const { actions, dispatch } = this.props;
        dispatch(actions.showAbout());
    }

    onNotFoundClick() {
        const { actions, dispatch } = this.props;
        dispatch(actions.showNotFound());
    }

    render() {
        return (
            <div className="layout-wrapper">
                <button onClick={e => this.onAboutClick(e)}>About</button>
                <button onClick={e => this.onNotFoundClick(e)}>404 Not Found</button>
                {this.props.children}
            </div>
        );
    }
}
export default withRouter(Layout);
