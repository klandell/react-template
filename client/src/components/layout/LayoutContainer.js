import React, { Component } from 'react';
import { connect } from 'react-redux';
import { withRouter } from 'react-router';
import { showAbout, showNotFound } from 'Actions/layoutActions';
import Layout from './Layout';

@connect(() => ({}), dispatch => ({
    actions: {
        showAbout,
        showNotFound,
    },
    dispatch,
}))
class LayoutContainer extends Component {
    onAboutClick = () => {
        const { actions, dispatch } = this.props;
        dispatch(actions.showAbout());
    }

    onNotFoundClick = () => {
        const { actions, dispatch } = this.props;
        dispatch(actions.showNotFound());
    }

    render() {
        const { onAboutClick, onNotFoundClick, props } = this;
        return (
            <Layout
              onAboutClick={onAboutClick}
              onNotFoundClick={onNotFoundClick}
            > {props.children}</Layout>
        );
    }
}
export default withRouter(LayoutContainer);
