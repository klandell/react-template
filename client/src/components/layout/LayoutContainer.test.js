import React from 'react';
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router';
import configureMockStore from 'redux-mock-store';
import LayoutContainer from './LayoutContainer';
import Layout from './Layout';

const mockStore = configureMockStore();

test('LayoutWrapper properly renders the layout component with no children.', () => {
    const store = mockStore();
    const component = mount(
        <Provider store={store}>
            <MemoryRouter>
                <LayoutContainer />
            </MemoryRouter>
        </Provider>,
    );
    const tree = toJson(component.find(Layout));
    expect(tree).toMatchSnapshot();
});

test('LayoutWrapper properly renders the layout component with children.', () => {
    const store = mockStore();
    const component = mount(
        <Provider store={store}>
            <MemoryRouter>
                <LayoutContainer>
                    <div>Foo</div>
                    <div>Bar</div>
                </LayoutContainer>
            </MemoryRouter>
        </Provider>,
    );
    const tree = toJson(component.find(Layout));
    expect(tree).toMatchSnapshot();
});

// TODO: test the event listeners
/*
test('Blah.....', () => {
    const store = mockStore();
    const component = mount(
        <Provider store={store}>
            <MemoryRouter>
                <LayoutContainer />
            </MemoryRouter>
        </Provider>,
    );

    const cmp = component.find(LayoutContainer).mount();

    // cmp.onAboutClick = jest.fn();

    component.find('.about-btn').simulate('click');

    console.log(cmp);
    // expect(cmp.onAboutClick).toBeCalled();
});
*/
