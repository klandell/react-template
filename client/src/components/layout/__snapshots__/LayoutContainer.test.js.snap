// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LayoutWrapper properly renders the layout component with children. 1`] = `
<Layout
  onAboutClick={[Function]}
  onNotFoundClick={[Function]}
>
  <div
    className="layout"
  >
    <button
      className="about-btn"
      onClick={[Function]}
    >
      About
    </button>
    <button
      className="not-found-btn"
      onClick={[Function]}
    >
      404 Not Found
    </button>
     
    <div>
      Foo
    </div>
    <div>
      Bar
    </div>
  </div>
</Layout>
`;

exports[`LayoutWrapper properly renders the layout component with no children. 1`] = `
<Layout
  onAboutClick={[Function]}
  onNotFoundClick={[Function]}
>
  <div
    className="layout"
  >
    <button
      className="about-btn"
      onClick={[Function]}
    >
      About
    </button>
    <button
      className="not-found-btn"
      onClick={[Function]}
    >
      404 Not Found
    </button>
     
  </div>
</Layout>
`;
